ui/pages/device_code.tsx

```diff
@@ -1,25 +1,83 @@
 import type { NextPage } from "next";
-import React from "react";
+import axios, { AxiosError } from "axios";
+import { useRouter } from "next/router";
+import React, { FormEvent, useCallback, useEffect, useState } from "react";
 import PageLayout from "../components/PageLayout";
-import { Button, Input } from "@canonical/react-components";
+import { Button, Form, Input } from "@canonical/react-components";
+
+export interface Response {
+  data: {
+    redirect_to?: string;
+  };
+}
+
+async function acceptUserCode(userCode: string, challenge: string) {
+  return axios
+    .put("/api/device?device_challenge=" + String(challenge), {
+      user_code: userCode,
+    })
+    .then(({ data }: Response) => {
+      if (data.redirect_to) {
+        window.location.href = data.redirect_to;
+      }
+    });
+}
 
 const DeviceCode: NextPage = () => {
+  const router = useRouter();
+  const { device_challenge: challenge, user_code: code } = router.query;
+  const [errorMessage, setErrorMessage] = useState("");
+
+  useEffect(() => {
+    // If the router is not ready yet, or we already have a flow, do nothing.
+    if (!router.isReady) {
+      return;
+    }
+  }, [code, errorMessage]);
```

Comment:
This is a no-op, can it be removed? Or should the content maybe not be wrapped in a use effect?

Reply:
I removed the `useEffect`, should be fine now